clang/lib/Lex/PPDirectives.cpp

```diff
@@ -177,6 +177,26 @@ static bool isLanguageDefinedBuiltin(const SourceManager &SourceMgr,
   return false;
 }
 
+static bool isReservedAttrName(Preprocessor &PP, IdentifierInfo *II) {
+  const LangOptions &Lang = PP.getLangOpts();
+  const StringRef Name = II->getName();
+
+  if (Lang.CPlusPlus26)
+    return Name == "indeterminate";
+  if (Lang.CPlusPlus23)
+    return Name == "assume";
```

Comment:
It seems that `#define assume` will not trigger diagnostic in C++26. Do I misunderstand this code?

Reply:
`Lang.CPlusPlus23` is set when the version is at least C++23.

Reply:
> It seems that `#define assume` will not trigger diagnostic in C++26.

I mean, control flow will reach line 185 and check if the name is `"indeterminate"` only, not check `"assume"`.

I think it shoule be
```cpp
if (Lang.CPlusPlus26 && Name == "indeterminate")
  return true;
```

Reply:
I'll revise this and add a test to cover this case.